hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientUserCodeDeploymentService.java

```diff
@@ -66,8 +67,37 @@ public void start() throws IOException, ClassNotFoundException {
         loadClasses();
     }
 
+    public static List<Class> sortByInheritance(List<Class<?>> classList) {
+        List<Class> sortedList = new LinkedList<Class>();
+        for (Class aClass : classList) {
+            boolean isInserted = false;
+
+            int index = 0;
+            for (Class sortedClass : sortedList) {
+                if (aClass.isAssignableFrom(sortedClass)) {
+                    sortedList.add(index, aClass);
+                    isInserted = true;
+                    break;
+                }
+                index++;
+            }
+            if (!isInserted) {
+                sortedList.add(aClass);
+            }
+        }
+        return sortedList;
+    }
+
+
     private void loadClasses() throws ClassNotFoundException {
-        for (String className : clientUserCodeDeploymentConfig.getClassNames()) {
+        List<String> classNames = clientUserCodeDeploymentConfig.getClassNames();
+        List<Class<?>> classes = new ArrayList<Class<?>>(classNames.size());
+        for (String className : classNames) {
+            classes.add(configClassLoader.loadClass(className));
+        }
+        List<Class> sortedClasses = sortByInheritance(classes);
```

Comment:
would this also do the same? 
```
        Collections.sort(classes, (c1, c2) -> c1.equals(c2) ? 0 : c1.isAssignableFrom(c2) ? -1 : 1);
```

Reply:
It is not a correct comparator implementation, because it does not give a total order. See this question and answer
https://stackoverflow.com/questions/50921813/how-to-sort-classes-according-to-their-inheritance-level#comment88888102_50922334